src/support/parallel_for.cc

```diff
@@ -93,5 +93,52 @@ void parallel_for(int begin, int end, const std::function<void(int)>& f, int ste
   }
 }
 
+void parallel_for_dynamic(int begin, int end, int num_threads,
+                          const std::function<void(int thread_id, int task_id)>& f) {
+  // Step 1. Sanity checks
+  if (begin == end) {
+    return;
+  }
+  CHECK_LT(begin, end) << "ValueError: The interval [begin, end) requires `begin <= end`";
```

Comment:
Should be `begin < end` right?

Reply:
Oh my bad. Thanks for spotting this!

Reply:
Oh wait actually it's correct. `begin == end` is allowed

Reply:
Then it should be `LE` and the interval should be `[begin, end]`. Would you confirm?

Reply:
the three lines above the check allows early exit if begin == end

Reply:
Ah, I got it. Thanks for the clarification. Just thinking `[begin, end)` and `begin <= end` might better be consistent.

Reply:
yeah it makes sense. Im going to change it to CHECK_LE to be more consistent